File: frontend/src/modules/graph/application/use-cases/readStatus.ts
import { StatusReader } from "@modules/graph/application/contracts/StatusReader";
import { WorkingStatus } from "@modules/graph/domain/workingStatus";

export async function readStatus(
  statusReader: StatusReader,
  repositoryPath: string
): Promise<WorkingStatus> {
  return statusReader.readStatus(repositoryPath);
}

File: backend/src/app/status/read_status.rs
use std::path::Path;

use crate::app::status::contracts::StatusReader;
use crate::domain::errors::AppError;
use crate::domain::working_status::WorkingStatus;

pub fn read_status(
    status_reader: &dyn StatusReader,
    repository_path: &Path,
) -> Result<WorkingStatus, AppError> {
    status_reader.read_status(repository_path)
}
